docs/src/main/sphinx/admin/properties.rst

```diff
@@ -21,3 +21,88 @@ may be used to tune Trino or alter its behavior when required.
     Logging <properties-logging>
     Web UI <properties-web-interface>
     Regular expression function <properties-regexp-function>
+
+Property value types
+--------------------
+
+Trino configuration properties support different value types with their own
+allowed values and syntax. Additional limitations apply on a per-property basis,
+and disallowed values result in a validation error.
+
+.. _prop-type-boolean:
+
+``boolean``
+^^^^^^^^^^^
+
+The properties of type ``boolean`` support two values, ``true`` or ``false``.
+
+.. _prop-type-data-size:
+
+``data size``
+^^^^^^^^^^^^^
+
+The properties of type ``data size`` support values that describe an amount of
+data, measured in byte-based units. These units are incremented in multiples of
+1024, so one megabyte is 1024 kilobytes, one kilobyte is 1024 bytes, and so on.
+For example, the value ``6GB`` describes six gigabytes, which is
+(6 * 1024 * 1024 * 1024) = 6442450944 bytes.
+
+The ``data size`` type supports the following units:
+
+* ``B``: Bytes
+* ``kB``: Kilobytes
+* ``MB``: Megabytes
+* ``GB``: Gigabytes
+* ``TB``: Terabytes
+* ``PB``: Petabytes
+
+.. _prop-type-double:
+
+``double``
+^^^^^^^^^^
+
+The properties of type ``double`` support numerical values including decimals,
+such as ``1.6``.
+
+.. _prop-type-duration:
+
+``duration``
+^^^^^^^^^^^^
+
+The properties of type ``duration`` support values describing an
+amount of time. For example, the value ``7m`` describes seven minutes.
+``duration`` type values must be non-negative.
+
+The ``duration`` type supports the following units:
+
+* ``ns``: Nanoseconds
+* ``us``: Microseconds
+* ``ms``: Milliseconds
+* ``s``: Seconds
+* ``m``: Minutes
+* ``h``: Hours
+* ``d``: Days
+
+A duration of ``0`` is treated as zero regardless of the unit that follows.
+For example, ``0s`` and ``0m`` both mean the same thing.
+
+.. _prop-type-integer:
+
+``integer``
+^^^^^^^^^^^
+
+The properties of type ``integer`` support whole numeric values, such as ``5``
+and ``1000``. Negative values are supported as well, for example ``-7``.
+``integer`` type values must be whole numbers, decimal values such as ``2.5``
+are not supported.
+
+Some ``integer`` type properties enforce their own minimum and maximum values.
+
+.. _prop-type-string:
+
+``string``
+^^^^^^^^^^
+
+The properties of type ``string`` support a set of values that consist of a
+sequence of characters. Allowed values are defined on a property-by-property
+basis, refer to the specific property for its supported and default values.
```

Comment:
any length restriction?

Reply:
multiline string possible? I think not but..

Reply:
It's a properties file - as long as your escape the value properly multiline strings will also be accepted (unless the validation for a property prevents it).
No length restrictions AFAIK.